src/pages/index.js

```diff
@@ -22,57 +24,117 @@ const profileDescriptionInput = document.querySelector(
+  headers: {
+    authorization: "9bd6fe3e-e63d-4619-b9fe-4bda3d127f2c",
+    "Content-Type": "application/json",
+  },
+});
+
+// let cardSection;
+
+// api
+//   .getUserInfo()
+//   .then((data) => {
+//     console.log(data);
+//     userInfo.setUserInfo(data);
+//     userInfo.setAvatar(data.modal);
+//   })
+//   .catch((err) => {
+//     console.log(err);
+//   });
+
+const previewImagePopup = new PopupWithImage("#preview-modal");
+previewImagePopup.setEventListeners();
+
 const cardList = new Section(
   {
-    items: initialCards,
     renderer: (item) => {
       const cardELement = createCard(item);
       cardList.addItem(cardELement);
     },
   },
   ".cards__list"
 );
-cardList.renderItems();
+api
+  .getInitialCards()
+  .then((cards) => {
+    console.log(cards);
+    cardList.renderItems(cards);
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
+const confirmDeleteModal = new PopupWithConfirm(
+  "#trashcan-modal",
+  handleDeleteClick
+);
+confirmDeleteModal.setEventListeners();
+
+const editAvatarModal = new PopupWithForm(
+  "#edit-avatar-modal",
+  handleAvatarEditSubmit
+);
+editAvatarModal.setEventListeners();
 
 const userInfo = new UserInfo({
   nameSelector: ".profile__title",
   descriptionSelector: ".profile__description",
+  avatarSelector: ".profile__image",
 });
 
 // const addCardModal = new PopupWithForm({
 //   popupSelector: "#add-card-modal",
 //   handleFormSubmit: handleAddCardFormSubmit,
 // });
-const addCardModal = new PopupWithForm(
-  selectors.addCardModalSelector,
-  handleAddCardFormSubmit
-);
+const addCardPopup = new PopupWithForm("#add-card-modal", (data) => {
+  const cardData = {
+    name: data.title,
+    link: data.url,
+  };
+
+  addCardPopup.setLoadingState(true);
 
-addCardModal.setEventListeners();
+  api
+    .createNewCard(cardData)
+    .then((res) => {
+      const card = createCard(res);
+      cardList.addItem(card);
+      addCardFormElement.reset();
+      addFormValidator.resetValidation();
+      addCardPopup.close();
+    })
+    .catch((err) => {
+      console.error(`Error, could not add card: ${err}`);
+    })
+    .finally(() => {
+      // addCardFormElement.reset();
+      addCardPopup.setLoadingState(false);
+    });
+});
+addCardPopup.setEventListeners();
 
 const editProfileModal = new PopupWithForm(
   "#profile-edit-modal",
   handleProfileEditSubmit
 );
-editProfileModal.setEventListeners();
 
-/* ------------------------------ event handler ----------------------------- */
+editProfileModal.setEventListeners();
+/* ------------------------------ event listeners ----------------------------- */
 cardAddButton.addEventListener("click", () => {
   addFormValidator.toggleButtonState();
-  addCardModal.open();
+  addCardPopup.open();
 });
 
 profileEditButton.addEventListener("click", () => {
   const formValues = userInfo.getUserInfo();
+  // editProfileModal.setLoadingState(true);
+  profileEditValidator.resetValidation();
```

Comment:
`  profileEditValidator.resetValidation();
`

* The profile modal doesn't open https://snipboard.io/TEDUPx.jpg 


`editFormValidator` should be here


`editFormValidator.resetValidation();`